static/js/tools/stat_var/page.tsx

```diff
@@ -167,12 +166,130 @@ class Page extends Component<unknown, PageStateType> {
     );
   }
 
-  private updateHash(sv: string): void {
-    window.location.hash = `#${sv}`;
+  /**
+   * Fetches sources to display in dropdown menu.
+   */
+  private async fetchSources(): Promise<void> {
+    axios
+      .get("/api/browser/propvals/typeOf/Source")
+      .then((resp) => {
+        const sourcePromises = [];
+        for (const source of resp.data?.values?.in) {
+          const url = SVG_URL_PREFIX + source.dcid;
+          sourcePromises.push(axios.get(url).then((resp) => resp));
+        }
+        if (sourcePromises.length === 0) {
+          return;
+        }
+        Promise.all(sourcePromises).then((sourceResults) => {
+          const sourceDcids = [];
+          for (const result of sourceResults) {
+            // Filter out all sources which have no stat vars in the main hierarchy (e.g. BMDC).
+            // TODO: Use ENTITY in schema to identify sources with stats
+            if (result.data.descendentStatVarCount) {
+              sourceDcids.push(
+                result?.config?.url.replace([SVG_URL_PREFIX], "")
+              );
+            }
+          }
+          if (sourceDcids.length === 0) {
+            return;
+          }
+          axios
+            .get(`/api/stats/propvals/name/${sourceDcids.join("^")}`)
+            .then((resp) => {
+              const sources = [];
+              for (const dcid in resp.data) {
+                sources.push({
+                  dcid,
+                  name: resp.data[dcid][0],
+                });
+              }
+              this.setState({ sources });
+            });
+        });
+      })
+      .catch(() => {
+        alert("Error fetching data.");
+      });
+  }
+
+  /**
+   * Updates entity for current source and dataset and also sets datasets based on source.
+   * @param source DCID of source
+   * @param dataset DCID of dataset
+   */
+  private async updateEntity(source: string, dataset: string): Promise<void> {
```

Comment:
same here, I don't think we need async & can just return void

Reply:
done